security/backends/elasticsearch/models.py

```diff
@@ -0,0 +1,566 @@
+    return ContentType.objects.db_manager(using).get_for_model(model)
+
+
+def get_key_from_content_type_and_id(content_type, object_id, model_db=None):
+    model_db = model_db or router.db_for_write(content_type.model_class())
+    return '|'.join((model_db, str(content_type.pk), str(object_id)))
+
+
+def get_key_from_object(obj, model_db=None):
+    return get_key_from_content_type_and_id(_get_content_type(obj), obj.pk, model_db)
+
+
+def get_object_from_key(key):
+    model_db, content_type_id, object_id = key.split('|')
+    return ContentType.objects.get(
+        pk=content_type_id
+    ).model_class().objects.using(model_db).filter(pk=object_id).first()
+
+
+def _get_response_state(status_code):
+    if status_code >= 500:
+        return RequestLogState.ERROR
+    elif status_code >= 400:
+        return RequestLogState.WARNING
+    else:
+        return RequestLogState.INFO
+
+
+logger_name_to_log_model = {
+    LoggerName.INPUT_REQUEST: InputRequestLog,
+    LoggerName.OUTPUT_REQUEST: OutputRequestLog,
+    LoggerName.COMMAND: CommandLog,
+    LoggerName.CELERY_TASK_INVOCATION: CeleryTaskInvocationLog,
+    LoggerName.CELERY_TASK_RUN: CeleryTaskRunLog,
+}
+
+
+def _get_log_model_from_logger_name(logger_name):
+    return logger_name_to_log_model[logger_name]
+
+
+def get_log_from_key(key):
+    logger_name, id = key.split('|')
+    return _get_log_model_from_logger_name(logger_name).get(id=id)
+
+
+def get_log_key(log):
+    logger_name = {v: k for k, v in logger_name_to_log_model.items()}[log.__class__]
+    return '{}|{}'.format(logger_name, log.meta.id)
+
+
+@receiver(input_request_started)
+def input_request_started_receiver(sender, logger, **kwargs):
+    related_objects = [
+        get_key_from_object(related_object) for related_object in logger.related_objects
+    ]
+
+    input_request_log = InputRequestLog(
+        slug=logger.slug,
+        extra_data=logger.extra_data,
+        state=RequestLogState.INCOMPLETE,
+        related_objects=related_objects,
+        **logger.data
+    )
+    input_request_log.meta.id = logger.id
+    if logger.parent_with_id:
+        input_request_log.parent_log = '{}|{}'.format(logger.parent_with_id.name, logger.parent_with_id.id)
+    input_request_log.save()
+
+
+@receiver(input_request_finished)
+def input_request_finished_receiver(sender, logger, **kwargs):
+    input_request_log = InputRequestLog.get(id=logger.id)
+    input_request_log.update(
+        slug=logger.slug,
+        extra_data=logger.extra_data,
+        state=_get_response_state(logger.data['response_code']),
+        refresh=settings.ELASTICSEARCH_AUTO_REFRESH,
+        **logger.data
+    )
+
+
+@receiver(input_request_error)
+def input_request_error_receiver(sender, logger, **kwargs):
+    input_request_log = InputRequestLog.get(id=logger.id)
+    input_request_log.update(
+        slug=logger.slug,
+        extra_data=logger.extra_data,
+        refresh=settings.ELASTICSEARCH_AUTO_REFRESH,
```

Comment:
refresh slow down write so by default is turned off. But it achive that the data will be in the response for the next read